nomad/drainer/watch_nodes.go

```diff
@@ -148,21 +148,22 @@ func NewNodeDrainWatcher(ctx context.Context, limiter *rate.Limiter, state *stat
 
 // watch is the long lived watching routine that detects node changes.
 func (w *nodeDrainWatcher) watch() {
+	// note: Try to avoid logging in this function, which is a background
+	// goroutine that is not waited on during a server shutdown. There is a race
+	// condition between logging and the testing.T being complete which triggers
+	// a panic, even if the test passes.
+
```

Comment:
You don't even need this. I think the only reason for the immense amount of logging here is that it was one of my first server-side contributions, and I was nervous. :sweat_smile: